src/Interface/Modules/Render/ViewScene.cc

```diff
@@ -2145,6 +2157,51 @@ void ViewSceneDialog::screenshotClicked()
   screenshotTaker_->saveScreenshot();
 }
 
+//--------------------------------------------------------------------------------------------------
+void ViewSceneDialog::reportBugClicked() {
+  std::string glVersion = reinterpret_cast<const char*>(glGetString(GL_VERSION));
+  std::string gpuVersion = reinterpret_cast<const char *>(glGetString(GL_RENDERER));
+
+  // Temporarily save screenshot so that it can be sent over email
+  takeScreenshot();
+  QImage image = screenshotTaker_->getScreenshot();
+  QString location = QDir::homePath() + QLatin1String("/scirun5screenshots/scirun_bug.png");
+  image.save(location);
+
+  // Generate email template
+  static std::string askForScreenshot = "\nIMPORTANT: Make sure to attach the screenshot of the ViewScene located at "
```

Comment:
I'd change all this to using QString rather than std::string, then use QStringBuilder operations to look cleaner, as in https://doc.qt.io/qt-5/qstring.html#more-efficient-string-construction

Reply:
Great tip. It should be a lot more efficient now.